java/arcs/sdk/testing/BaseTestHarness.kt

```diff
@@ -129,24 +129,15 @@ open class BaseTestHarness<P : Particle>(
         particle.onCreate()
 
         val readySoFar = atomic(0)
-        val readyJobs = handles.map { (name, handle) ->
+        val readyJobs = handles.map { (_, handle) ->
             launch {
-                // TODO: switch to this version when onReady is non-suspending:
-                // suspendCoroutine<Unit> { cont ->
-                //    handle.onReady {
-                //        val ready = readySoFar.incrementAndGet()
-                //        cont.resume(Unit)
-                //        launch { particle.onHandleSync(handle, ready == handles.size) }
-                //    }
-                // }
-                // TODO: get rid of this version when onReady is non-suspending
-                val job = Job()
-                handle.onReady {
-                    val ready = readySoFar.incrementAndGet()
-                    job.complete()
-                    launch { particle.onHandleSync(handle, ready == handles.size) }
+                suspendCoroutine<Unit> { cont ->
+                    handle.onReady {
+                        val ready = readySoFar.incrementAndGet()
+                        cont.resume(Unit)
+                        launch { particle.onHandleSync(handle, ready == handles.size) }
```

Comment:
We'll be able to remove this launch once onHandleSync is a non-suspending function.